### PR TITLE
first draft for adding github actions

### PR DESCRIPTION
**Pull request by etadewal**

### CODE REVIEW

File: .github/workflows/MINLP.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: MINLP
 on:
   push:
-    branches: [master]
+    branches: [minlp-test]
```
Comment (matbesancon) [inline .github/workflows/MINLP.yml:4]:
```suggestion
    branches: [master]
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        version: ['1.6', '1']
+        version: ['1.3', '1']  # Test against current minor release (at least 1.3)
```
Comment (matbesancon) [inline .github/workflows/ci.yml:14]: this is an error in the merge, should remain as is

Comment (etadewal) [reply .github/workflows/ci.yml:14]: I've changed it back to 1.6

File: .github/workflows/MINLP.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      fail-fast: false
+      matrix:
+        version: ['1.6', '1']  # Test against current minor release (at least 1.3)
+        os: [ubuntu-latest, macOS-latest]
+        arch: [x64]
+    steps:
+      - shell: bash
+        run: julia --project -e 'import Pkg; Pkg.instantiate(); Pkg.activate("SCIP"); Pkg.develop("SCIP")' 
+      - shell: bash
+        run: julia test/MINLPTests/run_minlptests.jl
```
Comment (matbesancon) [inline .github/workflows/MINLP.yml:21]:
```suggestion
        run: julia --project=test/MINLPTests test/MINLPTests/run_minlptests.jl

```

File: .github/workflows/ci.yml (hunk 2 above)
Comment (matbesancon) [reply .github/workflows/ci.yml:14]: it still appears here?

Comment (etadewal) [reply .github/workflows/ci.yml]: I made the same mistake in MINLP.yl and reverted it there but not here. Should be back to 1.6 in both now.

File: .github/workflows/MINLP.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        version: ['1.6', '1']  # Test against current minor release (at least 1.3)
+        os: [ubuntu-latest, macOS-latest]
+        arch: [x64]
+    steps:
+      - shell: bash
+        run: julia --project -e 'import Pkg; Pkg.instantiate(); Pkg.activate(); Pkg.develop("SCIP"); Pkg.activate("test/MINLPTests")' 
+      - shell: bash
+        run: cd SCIP.jl/test/MINLPTests
+      - shell: bash
+        run: julia --project=test/MINLPTests; test/MINLPTests/run_minlptests.jl
```
Comment (matbesancon) [inline .github/workflows/MINLP.yml:23]:
```suggestion
        run: julia --project=test/MINLPTests test/MINLPTests/run_minlptests.jl
```

**Comment by matbesancon** [comment]
@etadewal just made a change, I think the `cd test...` block above is not needed

File: .github/workflows/MINLP.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version: ['1.6', '1']  # Test against current minor release (at least 1.3)
+        os: [ubuntu-latest, macOS-latest]
+        arch: [x64]
+    steps:
+      - shell: bash
+        run: julia --project -e 'import Pkg; Pkg.instantiate(); Pkg.activate(); Pkg.develop("SCIP"); Pkg.activate("test/MINLPTests")'
```
Comment (matbesancon) [inline .github/workflows/MINLP.yml:19]:
```suggestion
        run: julia --project -e 'import Pkg; Pkg.instantiate(); Pkg.develop("."); Pkg.activate("test/MINLPTests"); Pkg.instantiate()' 
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        version: ['1.6', '1']
+        version: ['1.6', '1']  # Test against current minor release (at least 1.3)
```
Comment (matbesancon) [inline .github/workflows/ci.yml:14]:
```suggestion
        version: ['1.6', '1']
```

File: .github/workflows/MINLP.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version: ['1.6', '1']  # Test against current minor release (at least 1.3)
+        os: [ubuntu-latest, macOS-latest]
+        arch: [x64]
+    steps:
+      - shell: bash
+        run: julia --project -e 'import Pkg; Pkg.instantiate(); Pkg.develop(.); Pkg.activate("test/MINLPTests"); Pkg.instantiate()'
```
Comment (matbesancon) [inline .github/workflows/MINLP.yml:19]:
```suggestion
        run: julia --project -e 'import Pkg; Pkg.instantiate(); Pkg.develop("."); Pkg.activate("test/MINLPTests"); Pkg.instantiate()' 
```

File: .github/workflows/MINLP.yml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        with:
+          path: ~/.julia/artifacts
+          key: ${{ runner.os }}-test-${{ env.cache-name }}-${{ hashFiles('**/Project.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-test-${{ env.cache-name }}-
+            ${{ runner.os }}-test-
+            ${{ runner.os }}-
+      - uses: julia-actions/julia-buildpkg@v1
+      - shell: bash
+        run: julia --project -e 'import Pkg; Pkg.instantiate(); Pkg.develop("."); Pkg.activate("test/MINLPTests"); Pkg.instantiate()'
```
Comment (matbesancon) [inline .github/workflows/MINLP.yml:35]:
```suggestion
        run: julia --project -e 'import Pkg; Pkg.develop(PackageSpec(path=pwd())); Pkg.activate("test/MINLPTests"); Pkg.instantiate()' 
```

File: .github/workflows/MINLP.yml [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        with:
+          path: ~/.julia/artifacts
+          key: ${{ runner.os }}-test-${{ env.cache-name }}-${{ hashFiles('**/Project.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-test-${{ env.cache-name }}-
+            ${{ runner.os }}-test-
+            ${{ runner.os }}-
+      - uses: julia-actions/julia-buildpkg@v1
+      - shell: bash
+        run: julia --project -e 'import Pkg; Pkg.develop(PackageSpec(path=pwd())); Pkg.activate("test/MINLPTests"); Pkg.instantiate()'
```
Comment (matbesancon) [inline .github/workflows/MINLP.yml:35]:
```suggestion
        run: julia --project -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd())); Pkg.activate("test/MINLPTests"); Pkg.instantiate()' 
```

File: .github/workflows/MINLP.yml [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+        with:
+          path: ~/.julia/artifacts
+          key: ${{ runner.os }}-test-${{ env.cache-name }}-${{ hashFiles('**/Project.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-test-${{ env.cache-name }}-
+            ${{ runner.os }}-test-
+            ${{ runner.os }}-
+      - uses: julia-actions/julia-buildpkg@v1
+      - shell: bash
+        run: julia --project -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd())); Pkg.activate("test/MINLPTests"); Pkg.instantiate()'
```
Comment (matbesancon) [inline .github/workflows/MINLP.yml:35]:
```suggestion
        run: julia -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd())); Pkg.activate("test/MINLPTests"); Pkg.instantiate()'
```

**Pull request merged**